service/worker/common/interface.go

```diff
@@ -40,4 +40,22 @@ type (
 		TaskQueue string
 		Options   sdkworker.Options
 	}
+
+	// PerNamespaceWorkerComponent represent a type of work needed for worker role
+	PerNamespaceWorkerComponent interface {
+		// Register registers Workflow and Activity types provided by this worker component.
+		Register(sdkworker.Worker)
+		// DedicatedWorkerOptions returns a DedicatedWorkerOptions for this worker component.
+		// Must not return nil.
+		DedicatedWorkerOptions() *PerNamespaceDedicatedWorkerOptions
+	}
+
+	PerNamespaceDedicatedWorkerOptions struct {
+		// TaskQueue must be present, there is no default task queue
+		TaskQueue string
+		// How many worker nodes should run a worker per namespace
+		NumWorkers int
+		// Other worker options
+		Options sdkworker.Options
+	}
```

Comment:
Can you add NumWorkers to DedicatedWorkerOptions and reuse the existing WorkerComponent / DedicatedWorkerOptions without creating these 2 new ones? You can use named group for fx to know each one belong to different group.

Reply:
My thinking was that the semantics are already slightly different, and the options might diverge more in the future, so it would be good to have separate types. But I guess the whole thing may just be a stop-gap so it's fine.